pkg/commands/compute/deploy.go

```diff
@@ -91,330 +84,215 @@ func (c *DeployCommand) Exec(in io.Reader, out io.Writer) (err error) {
-			if err != nil {
-				c.Globals.ErrLog.AddWithContext(err, map[string]interface{}{
-					"Service ID":      serviceID,
-					"Service Version": version.Number,
-				})
-				return fmt.Errorf("error cloning service version: %w", err)
-			}
-			if c.Globals.Flag.Verbose {
-				msg := fmt.Sprintf("Service version %d is not editable, so it was automatically cloned. Now operating on version %d.", version.Number, v.Number)
-				text.Break(out)
-				text.Output(out, msg)
-				text.Break(out)
-			}
-			version = v
-		}
+	// Because a service_id exists in the fastly.toml doesn't mean it's valid
+	// e.g. it could be missing required resources such as a domain or backend.
+	// We check and allow the user to configure these settings before continuing.
+
+	hasDomain, err := serviceHasDomain(serviceID, serviceVersion.Number, apiClient)
+	if err != nil {
+		errLogService(errLog, err, serviceID, serviceVersion.Number)
+		return err
+	}
+
+	availableBackends, err := apiClient.ListBackends(&fastly.ListBackendsInput{
+		ServiceID:      serviceID,
+		ServiceVersion: serviceVersion.Number,
+	})
+	if err != nil {
+		errLogService(errLog, err, serviceID, serviceVersion.Number)
+		return fmt.Errorf("error fetching service backends: %w", err)
+	}
 
-		// We define the `ok` variable so that the following call to
-		// `validateservice` will be able to shadow `invalidType`.
-		var ok bool
+	var (
+		backendsToCreate    []Backend
+		domainToCreate      string
+		hasRequiredBackends bool
+		missingBackends     []manifest.Mapper
+	)
 
-		// Because a service_id exists in the fastly.toml doesn't mean it's valid
-		// e.g. it could be missing either a domain or backend resource. So we
-		// check and allow the user to configure these settings before continuing.
-		ok, invalidType, backendsMissing, err = validateService(serviceID, c.Globals.Client, version, c.Manifest.File.Setup.Backends)
+	predefinedBackends := c.Manifest.File.Setup.Backends
+	if len(predefinedBackends) > 0 {
+		missingBackends, err = serviceMissingConfiguredBackends(serviceID, serviceVersion.Number, apiClient, availableBackends, predefinedBackends)
 		if err != nil {
-			c.Globals.ErrLog.AddWithContext(err, map[string]interface{}{
-				"Service ID":      serviceID,
-				"Service Version": version.Number,
-			})
+			errLogService(errLog, err, serviceID, serviceVersion.Number)
 			return err
 		}
+		if len(missingBackends) == 0 {
+			hasRequiredBackends = true
+		}
+	} else if len(availableBackends) > 0 {
+		hasRequiredBackends = true
+	}
 
-		if !ok {
-			invalidService = true
+	// RESOURCE CONFIGURATION...
 
-			text.Output(out, "Service '%s' is missing required domain or backend(s). These must be added before the Compute@Edge service can be deployed.", serviceID)
+	if !hasDomain || !hasRequiredBackends {
+		if !c.AcceptDefaults {
+			text.Output(out, "Service '%s' is missing required resources. These must be added before the Compute@Edge service can be deployed. Please follow the prompts.", serviceID)
```

Comment:
If initialising a new service and then changing the host of a backend on manage.fastly.com this error will appear on subsequent deploys - suggest modifying fastly.toml file